jdaviz/utils.py

```diff
@@ -351,15 +377,66 @@ def flux_conversion(spec, values, original_units, target_units):
             eqv_in = [min(fac), max(fac)]
         else:
             eqv_in = fac
-
         eqv += _eqv_pixar_sr(np.array(eqv_in))
 
+        # transitive units cannot be directly converted, and require
+        # additional conversions to reach the desired end unit.
+        # if spec_unit in [original_units, target_units]:
+        values, updated_units, unit_type = _transitive_conversion(
+            values=values, orig_units=orig_units, targ_units=targ_units,
+            eqv=eqv, spec_unit=spec_unit
+            )
+        if unit_type and unit_type == 'targ':
+            targ_units = updated_units
+        elif unit_type and unit_type == 'orig':
+            orig_units = updated_units
+
+    elif image_data:
+        values, orig_units = _transitive_conversion(
+            values=values, orig_units=orig_units, targ_units=targ_units,
+            eqv=eqv, image_data=image_data
+            )
+
     return (values * orig_units).to_value(targ_units, equivalencies=eqv)
 
 
-def _convert_surface_brightness_units(data, from_unit, to_unit):
-    quantity = data * u.Unit(from_unit)
-    return quantity.to_value(u.Unit(to_unit))
+def _transitive_conversion(values, orig_units, targ_units, eqv, spec_unit=None, image_data=None):  # noqa
```

Comment:
"transitive" seems too vague. In unit conversion, you either can convert or not; sometimes you need extra equivalencies but astropy does not call them "transitive". In the guts of this function, it still focuses on surface brightness and flux, so `_convert_surface_brightness_units` might be better suited.

And that said, why do we need a `_convert_surface_brightness_units` with only 2 lines below?

Reply:
That works for me, I was trying to emphasizing that it was as straightforward as going to `A -> D`, but` A -> B -> C -> D` for specific units but I have no loyalty to the naming convention here so and good with `_convert_surface_brightness_units`. I kept the 2 line `_convert_surface_brightness_units` as a fallback for the pixel value conversions in case the pixel scale factor isn't present, so the 'translatable' units in surface brightness can still be converted in the case of image viewers.

I think both should be kept and overload the `_convert_surface_brightness_units`, but let me know if there's better practice.